static/js/Misc.jsx

```diff
@@ -1203,45 +1222,64 @@ const ReorderEditorWrapper = ({toggle, type, data}) => {
-      };
-      origData.origDesc = data.description || {"en": "", "he": ""};
-      origData.origCategoryDesc = data.categoryDescription || {"en": "", "he": ""};
-      const origWasCat = "displays-above" in data?.links;
-      return <TopicEditor origData={origData}
-                                    origWasCat={origWasCat}
-                                    onCreateSuccess={(slug) => window.location.href = "/topics/" + slug}
-                                    close={toggle}/>;
-  }
-  const getEditorForExistingCategory = () => {
-      let tocObject = Sefaria.tocObjectByCategories(data);
-      const origDesc = {en: tocObject.enDesc, he: tocObject.heDesc};
-      const origCategoryDesc = {en: tocObject.enShortDesc, he: tocObject.heShortDesc};
-      const origData = {
-        origEn: tocObject.category,
-        origHe: tocObject.heCategory,
-        origDesc,
-        origCategoryDesc,
-        isPrimary: tocObject.isPrimary
-      };
-      return <CategoryEditor origData={origData} close={toggle} origPath={data.slice(0, -1)}/>;
-  }
+const EditorForExistingTopic = ({ toggle, data }) => {
+  const initCatSlug = TopicToCategorySlug(data);
+  const origData = {
+    origSlug: data.slug,
+    origCategorySlug: initCatSlug,
+    origEn: data.primaryTitle.en,
+    origHe: data.primaryTitle.he || "",
+    origDesc: data.description || {"en": "", "he": ""},
+    origCategoryDesc: data.categoryDescription || {"en": "", "he": ""},
+  };
+  
+  const origWasCat = "displays-above" in data?.links;
+  
+  return (
+    <TopicEditor 
+      origData={origData}
+      origWasCat={origWasCat}
+      onCreateSuccess={(slug) => window.location.href = `"/topics/"${slug}`}
+      close={toggle}
+    />
+  );
+};
+
+
+
+const EditorForExistingCategory = ({ toggle, data }) => {
+  let tocObject = Sefaria.tocObjectByCategories(data);
+  const origDesc = {en: tocObject.enDesc, he: tocObject.heDesc};
+  const origCategoryDesc = {en: tocObject.enShortDesc, he: tocObject.heShortDesc};
+  const origData = {
+    origEn: tocObject.category,
+    origHe: tocObject.heCategory,
+    origDesc,
+    origCategoryDesc,
+    isPrimary: tocObject.isPrimary
+  };
 
+  return (
+    <CategoryEditor 
+      origData={origData} 
+      close={toggle} 
+      origPath={data.slice(0, -1)}
+    />
+  );
+};
+
+
+const CategoryEditorWrapper = ({toggle, data, type}) => {
   switch (type) {
-    case 'books':
+    case "books":
       return <EditTextInfo initTitle={data} close={toggle}/>;
     case "sources":
         const [topicSlug, refData] = data;
         return <SourceEditor topic={topicSlug} origData={refData} close={toggle}/>;
     case "cats":
-        return getEditorForExistingCategory();
+        return <EditorForExistingCategory toggle={toggleFunction} data={dataObject} />;
     case "topics":
-        return getEditorForExistingTopic();
+        return <EditorForExistingTopic toggle={toggleFunction} data={dataObject} />;
```

Comment:
it seemed that we could stay consistent here in what CategoryEditorWrapper returns, if we are expecting a functional component, it is cleaner to have that be the case across all conditions